src/web/util.ts

```diff
@@ -77,3 +79,15 @@ export const getWorkspaceRelativePath = (folderUri: vscode.Uri, fileUri: vscode.
 export type ArrayElement<ArrayType extends readonly unknown[]> = ArrayType extends readonly (infer ElementType)[] ? ElementType : never;
 
 export const keysForEnum = <M extends Record<string, unknown>>(map: M): [keyof M, ...(keyof M)[]] => Object.keys(map) as unknown as [keyof M, ...(keyof M)[]];
+
+export const offerSaveFile = async (content: String, options: vscode.SaveDialogOptions): Promise<vscode.Uri | undefined> => {
+    let chosenUri = await vscode.window.showSaveDialog(options);
+    if (!chosenUri){
+        return;
+    }
+
+    let buf = Buffer.from(content, 'utf8');
```

Comment:
The rest of the code base uses `encodeText` from `src/util.ts`, so I think we should be consistent.